### DISCUSSION TITLE
Risk number

### DISCUSSION THREAD

**Discussion by LauraGOP** [monarc-project/MonarcAppFO#419] [Ideas]
Hello,

  I have the latest version of the Monarc instance but I was wondering it was
  possible to have a new feature.

  We would like to have a column on the analysis to assign each risk a number
  (a sort of ID), which would make it easier to link the risks to the SoA.

  Is this something feasible? Something already in the pipeline?

  Kr,

**Comment by K8Sudo** [comment]
Hi LaraGOP,

Where do you need a reference? Have you seen that the SoA automatically links to risks according to the knowledge base?

![Monarc SoA Ref_1](https://user-images.githubusercontent.com/47662401/157200581-4c5790d8-ba42-4567-be95-9302d4df446f.png)

When you click on the reference. (marked with a rectangular box)
![Monarc SoA Ref_2](https://user-images.githubusercontent.com/47662401/157200624-4bfebf26-2829-445c-8736-1fc3bfa34707.png)

Does this help already?

Kind regards

**Comment by LauraGOP** [comment]
Hello,

Actually we would 
![1](https://user-images.githubusercontent.com/101174751/158348692-93ef8bfa-8d57-44ff-b6cc-38a9d9c2a15c.png)
like to have a number for each risk and each recommendation. See screenshot attached.

Kr,

**Comment by ruslanbaidan** [comment]
Hello,

We don't have the plan in the nearest future to do this.
Could you explain please, what is the benefit for you to have the numbers?

From my point of view, if you need to link risks to the SOA in the knowledge base, you could open multiple browser tabs in a separate window to be sure you are linked the right risks (image above). Probably 3 windows separate windows (all with the same session).
Numbers can also confuse or/and add an extra complexity.

Regards
![DoubleWindows](https://user-images.githubusercontent.com/3246171/159692346-cdea6d3d-b279-4806-9e40-6bd0beac75d0.png)

**Reply by K8Sudo** [reply]
We have also become aware of this problem more frequently in the meantime. A unique risk ID would help to ensure better documentation during team meetings, management reviews and audits.

**Reply by ruslanbaidan** [reply]
There is a unique integer risk ID in the database that is linked to each risk that is added to the analysis. It can be added to the lists where necessary.
Each of the informational risks in the knowledge base also has a unique identifier, but in UUID string format and 36 characters long.

**Reply by K8Sudo** [reply]
Hey @ruslanbaidan , 

yes, I had already noticed that. It would be cool if it was already included in the standard delivery of Monarc and didn't require customizing. At the very least, it's a topic that is always raised and requested in our quarterly training sessions.

**Reply by ruslanbaidan** [reply]
Hey @KonzeptAcht ,

I agree that an ID displaying could be a useful feature for Monarc users.
I would like to remind that Monarc is an open source tool that means customers, community members or partners are welcome to help with the development of new features, functionality or improving the existing ones.

The BackEnd part of the project is going to be refactored and it would be easier and clear to perform the changes or improvements.

I'm available to help on the questions related to the code modifications, review and accept pull requests.

**Reply by K8Sudo** [reply]
Hello @ruslanbaidan , 

thanks for the info. That wasn't meant to be an expectation at all. In the past, you have had many of these features on the roadmap yourself, so it wouldn't make sense for us to work on a solution at the same time as you. If you are not currently planning an integration, I will take this on board and will address and plan the topic in our team meeting next week.

**Comment by sfred92** [comment]
Hello to everyone,

+1 for me too to have an ID number for each identified risk.

Another example would be to identify a risk in an action plan, and not only its recommandation which has an ID. 

The display could be optional on the web interface. And we could choose a prefix (RAxx), that would be even better.

Thank you
Fred

**Comment by ruslanbaidan** [comment]
It was implemented in the v2.13.2.
